### PR TITLE
Merge v2.1.1 into `master`

### PR DESCRIPTION
**Pull request by MatthewDaggitt**
This brings across all the changes on `experimental` and `v2.1.1-release` branches onto `master`.  Therefore after this is merged everyone working on the library will need to update to v2.7.0 of Agda.

### CODE REVIEW

**Comment by jamesmckinna** [comment]
> This brings across all the changes on `experimental` and `v2.1.1-release` branches onto `master`. Therefore after this is merged everyone working on the library will need to update to v2.7.0 of Agda.

Ahead of a proper review, I have been testing recent PRs against both v2.6.4.3 and v2.7.0, so hopefully any eventual merge of this PR will be a painless transition... ;-)

File: .github/workflows/ci-ubuntu.yml
Comment (jamesmckinna) [inline .github/workflows/ci-ubuntu.yml:79]: I haven't verified that these are the correct commit hashes!
(But, for info: how could/should I, were I to want to?)

File: doc/installation-guide.md
Comment (jamesmckinna) [inline doc/installation-guide.md:18]: Is it worth (redundantly) repeating here the caveat "subject to compatibility with the relevant version of Agda", as above?

File: src/Reflection/AST/Show.agda
Comment (jamesmckinna) [inline src/Reflection/AST/Show.agda:64]: I realise I haven't kept up with the PRs on `experimental`, so I assume that these are the agreed `String` renderings... but they maybe seem a bit verbose?

**Review by jamesmckinna — APPROVED** [review]
This all looks fine, modulo the structure of `CHANGELOG` and some of the additional textual suggestions, but none is (I think!) a deal-breaker.

My only thought is that we perhaps try to clear the other remaining non-DRAFT open stdlib-v2.2/agda-v2.6.4.3 PRs (or else: as much of the v2.2 milestone as we can agree upon ;-)) before merging this one, in the interests of marking a clean 'break' between Agda versions? Or, at least, that we discuss this among the maintainer core team before proceeding (ie. observe more than the '2 approvals' rule)?

**Comment by jamesmckinna** [comment]
Consider also rolling the typo change `LICENSE` -> `LICENCE` in `doc/release-guide.txt` into this PR?

**Comment by andreasabel** [comment]
@MatthewDaggitt @jamesmckinna  Would be great the see this being merged quickly, because:
- For the upcoming 2.7.0.1 bugfix release of Agda I would like to track the `master` again rather than `experimental`.
- I have a new small commit for `experimental` concerning Agda 2.8.0.  (It is actually so small it can go into `master` as well.)  However, it is probably smarter to not touch `experimental` why this PR here is pending.

Or maybe I am wrong here and you can advise me a better workflow.

**Comment by jamesmckinna** [comment]
@andreasabel I'm not trying to get in the way, indeed I'm happy to go ahead and merge, but currently waiting on a second 'approve' review, and feedback from @MatthewDaggitt on my own round of comments. I guess we can take the first one as read from you (but please do so explicitly), and once Matthew comes back, happy to proceed... Hope this helps!

**Review by andreasabel — APPROVED** [review]

File: src/Reflection/AST/Show.agda
Comment (jamesmckinna) [reply src/Reflection/AST/Show.agda:64]: Happy to leave them as-is... the style for `show` operations does seem to favour a more verbose=informative style.

File: CHANGELOG/v2.1.1.md
Comment (jamesmckinna) [reply CHANGELOG/v2.1.1.md:6]: I'll leave this to @MatthewDaggitt to fix up for the 'actual' release, in the interests of merging sooner rather than later.

File: CITATION.cff
Comment (jamesmckinna) [reply CITATION.cff:7]: I'll update it to today for now, but as above, I'll leave this to @MatthewDaggitt to fix up for the 'actual' release, in the interests of merging sooner rather than later.

Comment (jamesmckinna) [inline CITATION.cff:7]:
```suggestion
date-released: 2024-09-07
```

**Comment by jamesmckinna** [comment]
Two cosmetic patches applied after @andreasabel 's comments, and will merge now.

**Pull request merged**
